EssentialsDiscord/src/main/java/net/essentialsx/discord/listeners/BukkitListener.java

```diff
@@ -104,7 +107,7 @@ public void onChat(AsyncPlayerChatEvent event) {
     @EventHandler(priority = EventPriority.MONITOR)
     public void onJoin(AsyncUserDataLoadEvent event) {
         // Delay join to let nickname load
-        if (event.getJoinMessage() != null) {
+        if (event.getJoinMessage() != null && (!jda.getSettings().isVanishHideMessages() || !event.getUser().isHidden())) {
```

Comment:
Maybe pull the new check out to a `boolean shouldShowJoinQuitMessage(User user)` method (or with a better name, maybe `shouldShowActivityMessage` so can be used later below)

Could maybe even put this in API - "by default" in the config comment implies other message types could respect this, and possibly worth exposing more easily to 3rd-party plugins, but it's up to you